htlcswitch/link.go

```diff
@@ -2387,6 +2401,45 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		// Update the mailbox's feerate as well.
 		l.mailBox.SetFeeRate(fee)
 
+	case *lnwire.Stfu:
+		handleErr := func(err error) {
+			l.log.Error(err.Error())
+			l.fail(
+				LinkFailureError{
+					code:             ErrInvalidUpdate,
+					FailureAction:    LinkFailureDisconnect,
+					PermanentFailure: false,
+					Warning:          true,
+				},
+				err.Error(),
+			)
+		}
+
+		remoteOnLocal := l.channel.NumRemoteUpdatesPendingOnLocal()
+		remoteOnRemote := l.channel.NumRemoteUpdatesPendingOnRemote()
```

Comment:
Based on the usage I think all we want to know if `HasRemoteUpdatesPending` and `HasLocalUpdatePending`?

Reply:
I can add convenience methods on top of these, but I do think these are broadly useful for the channel API overall

Reply:
I've consolidated the APIs down to the most useful primitive I can imagine.